settings/src/components/email-address.js

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * WordPress dependencies
+ */
+import { Button, TextControl, Notice, Spinner } from '@wordpress/components';
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * Render the Email setting.
+ */
+export default function EmailAddress( { userRecord } ) {
+	const { record, edit, save, editedRecord, hasEdits, isSaving } = userRecord
+	const [ emailError, setEmailError ]                            = useState( '' );
+	const [ justChangedEmail, setJustChangedEmail ]                = useState( false );
+
+	const handleSave = useCallback( async () => {
+		try {
+			setEmailError('');
+
+			await save();
+
+			setJustChangedEmail( true );
+		} catch( error ) {
+			let message = error.message;
+
+			if ( 'rest_user_invalid_email' === error.code ) {
+				message = 'Sorry, that email is already in use by another account.';
+			} else if ( 'rest_invalid_param' === error.code ) {
+				message = 'Sorry, it looks like that email is invalid.';
+			}
+
+			setEmailError( message );
+		}
+	}, [] );
+
+	const handleDiscard = useCallback( async () => {
+		try {
+			await edit( { pending_email: '' } );
+			await save();
```

Comment:
It might be a little safer to do something like `edit( record )` (and then not call `save()`. That'd guarantee that nothing is changed, and avoid an HTTP call.

I haven't tested if that actually works, though.

Reply:
Er, nevermind, this needs to `save()` to change the pending address in the db 👍🏻